integration/src/main/java/scripts/utils/ControlledAzureVmTestScriptBase.java

```diff
@@ -0,0 +1,223 @@
+        new AzureNetworkCreationParameters()
+            .name(String.format("network-%s", suffix))
+            .subnetName(String.format("subnet-%s", suffix))
+            .addressSpaceCidr("192.168.0.0/16")
+            .subnetAddressCidr("192.168.1.0/24")
+            .region(REGION);
+    networkRequestBody.azureNetwork(networkParameters);
+    CreatedControlledAzureNetwork network =
+        azureApi.createAzureNetwork(networkRequestBody, getWorkspaceId());
+    assertNotNull(network.getResourceId());
+    logger.info(
+        "Created Network with following parameters Name={}, Resource Id={}",
+        network.getAzureNetwork().getAttributes().getNetworkName(),
+        network.getResourceId());
+    return network;
+  }
+
+  protected CreatedControlledAzureDisk createDisk(String resourceSuffix) throws ApiException {
+    CreateControlledAzureDiskRequestBody diskRequestBody =
+        new CreateControlledAzureDiskRequestBody()
+            .common(createCommonFields("common-disk", resourceSuffix));
+    AzureDiskCreationParameters diskParameters =
+        new AzureDiskCreationParameters()
+            .name(String.format("disk-%s", suffix))
+            .size(50)
+            .region(REGION);
+    diskRequestBody.azureDisk(diskParameters);
+    CreatedControlledAzureDisk disk = azureApi.createAzureDisk(diskRequestBody, getWorkspaceId());
+    assertNotNull(disk.getResourceId());
+    logger.info(
+        "Created Disk with following parameters Name={}, Resource Id={}",
+        disk.getAzureDisk().getAttributes().getDiskName(),
+        disk.getResourceId());
+    return disk;
+  }
+
+  protected CreatedControlledAzureVmResult waitForVmCreationCompletion(
+      String createVmJobId, CreatedControlledAzureVmResult vmCreateResult)
+      throws InterruptedException, ApiException {
+    while (ClientTestUtils.jobIsRunning(vmCreateResult.getJobReport())) {
+      TimeUnit.SECONDS.sleep(VM_CREATE_WAIT_POLL_INTERVAL_SECONDS);
+      vmCreateResult = azureApi.getCreateAzureVmResult(getWorkspaceId(), createVmJobId);
+    }
+
+    return vmCreateResult;
+  }
+
+  protected ControlledResourceCommonFields createCommonFields(String name, String suffix) {
+    return new ControlledResourceCommonFields()
+        .name(String.format("%s-%s", name, suffix))
+        .cloningInstructions(CloningInstructionsEnum.NOTHING)
+        .accessScope(AccessScope.PRIVATE_ACCESS)
+        .managedBy(ManagedBy.USER);
+  }
+
+  protected AzureVmImage createAzureVmImage() {
+    return new AzureVmImage()
+        .publisher("microsoft-dsvm")
+        .offer("ubuntu-1804")
+        .sku("1804-gen2")
+        .version("22.04.27");
+  }
+
+  protected AzureVmUser createUser() {
+    String userName = "jupuser";
+    String userPassword = "SuperStrongPassword12345!!";
+    return new AzureVmUser().name(userName).password(userPassword);
+  }
+
+  protected AzureVmCustomScriptExtension createVmCustomScriptExtension() {
```

Comment:
nit: might be worth a comment that this is not added to a VM by default (unlike the other `create___` functions here which are used automatically)